x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -653,6 +654,28 @@ private Join resolveLookupJoin(LookupJoin join) {
                     return join;
                 }
 
+                // joining with remote cluster is not supported yet
+                if (join.left() instanceof EsRelation esr) {
+                    for (var index : esr.index().concreteIndices()) {
+                        if (index.indexOf(RemoteClusterAware.REMOTE_CLUSTER_INDEX_SEPARATOR) != -1) {
```

Comment:
I'd check both left() and right(), just to avoid the grammar being "fixed" and opening this possibility.
Also, there's `StringUtils#splitQualifiedIndex()` that helps with this check, but arguably heavier and maybe not absolutely required.
Both notes as FYI.

Reply:
I think the right part of the join can be checked in LogicalPlanBuilder where the `lookup` is built.

Reply:
In the current state, I think I'd prefer adding a check right here like Bogdan suggests. But, we could have both for maximum paranoia!

- Check in LogicalPlanBuilder: _currently_ redundant, as the grammar doesn't allow for `:` at all. I'll likely change my mind once we update the grammar. Because if we generalize the latter, a check in LogicalPlanBuilder would require only the parsed query; as such, it can prevent even sending a field caps request in case a user asks for `... | LOOKUP JOIN remote:lu_idx ...`.
- Check here, in the Analyzer: using the concrete indices is a good source of definitive truth, as the concrete indices cannot contain aliases.

Reply:
This is not about paranoia, but about performance: fail fast without even checking the field_caps call(s) or resolve the clusters/indices and then fail in the Analyzer.

This needs a bit of research... there may be situations (at the Verifier level) where the behavior might change if we check things too early. For example, `lookup join some_inexistent_remote_cluster:foo` - do we complain that a remote lookup index is not supported or that `some_inexistent_remote_cluster`  doesn't actually exist and the query is incorrect because maybe the user had a typo?